backend/src/game/game.service.ts

```diff
@@ -43,133 +42,138 @@ export class GameService {
       const user = await this.prisma.user.findUnique({
         where: {
           email: otherPlayer.email,
-        }
-      })
-      if (user.status !== 'WAITING'){
+        },
+      });
+      if (user.status !== 'WAITING') {
         this.cacheManager.del('pendingPlayer');
         pendingPlayer = undefined;
       }
-			if (otherPlayer.userName === player.userName){
-				this.cacheManager.set('pendingPlayer', JSON.stringify(player));
-				return [false, player.gameID];
-			}
+      if (otherPlayer.userName === player.userName) {
+        this.cacheManager.set('pendingPlayer', JSON.stringify(player));
+        return [false, player.gameID];
+      }
     }
 
     // Matching
-    if (!pendingPlayer && player){
-			this.createWaitingGame(player);
+    if (!pendingPlayer && player) {
+      this.createWaitingGame(player);
       this.cacheManager.set('pendingPlayer', JSON.stringify(player));
       return [false, player.gameID];
     } else {
       const otherPlayer = JSON.parse(pendingPlayer);
       this.cacheManager.del('pendingPlayer');
       const gameID = otherPlayer.gameID;
-			this.joinGameAndLaunch(player, gameID);
+      this.joinGameAndLaunch(player, gameID);
       return [true, otherPlayer.gameID];
     }
   }
 
-	async createWaitingGame(player: Player){
-		const gameID = this.gameIDcounter;
-		this.gameIDcounter++;
-		const newGame = new Game(
-			gameID,
-			1,
-			player,
-			null,
-			[10, 10],
-			{
-				x: 400,
-				y: 400,
-			},
-			{
-				x: 7,
-				y: 7,
-			},
-			Math.random() * Math.PI * 2,
-			GameStatus.Waiting,
-		);
-		player.gameID = gameID;
-		await this.prisma.user.update({
-			where: {
-				email: player.email
-			}, data : {
-				status: 'WAITING',
-			},
-		})
-		this.games.push(newGame);
-		console.log('game created with id', gameID);
-	}
-
-	async joinGameAndLaunch(player: Player, gameID: number) : Promise<boolean> {
-		const game = this.games.find((game) => game.gameID === gameID);
-		if (game){
-			player.gameID = gameID;
-			game.rightPlayer = player;
-			await this.prisma.user.updateMany({
-				where: {
-					email: {
-						in: [game.leftPlayer.email, game.rightPlayer.email],
-					},}, data : {
-					status: 'PLAYING',
-				},
-			})
-			game.status = GameStatus.Playing;
-			return true;
-		} else {
-			return false;
-		}
-	}
+  async createWaitingGame(player: Player){
+    const gameID = player.socketID;
+    const newGame = new Game(
+      gameID,
+      1,
+      player,
+      null,
+      [8, 8],
+      { x: 400, y: 400, },
+      { x: 5, y: 5, },
+      Math.random() * Math.PI * 2,
+      GameStatus.Waiting,
+    );
+    player.gameID = gameID;
+    try {
+      await this.prisma.user.update({
+        where: {
+          email: player.email,
+        },
+        data: {
+          status: 'WAITING',
+        },
+      });
+    } catch (err) {
+      console.log(err);
+    }
+    this.games.push(newGame);
+    console.log('game created with id', gameID);
```

Comment:
Shouldn't the game be stored in the cache?

Reply:
Yeh. Not started yet.... XD